IPython/core/magics/basic.py

```diff
@@ -294,13 +294,9 @@ def profile(self, parameter_s=''):
         prun : run code using the Python profiler
                (:meth:`~IPython.core.magics.execution.ExecutionMagics.prun`)
         """
-        warn("The `%profile` magic has been deprecated since IPython 2.0. Please use "
-             "the value of `get_ipython().profile` instead to see current profile in use.")
-        from IPython.core.application import BaseIPythonApplication
-        if BaseIPythonApplication.initialized():
-            print(BaseIPythonApplication.instance().profile)
-        else:
-            error("profile is an application-level value, but you don't appear to be in an IPython application")
+        raise DeprecationWarning("The `%profile` magic has been deprecated since IPython 2.0. "
```

Comment:
Do you mean `warnings.warn(...)`? Raise gives a traceback, which I don't think we want.